### DISCUSSION TITLE
Need help for understand the "alias" variables

### DISCUSSION THREAD

**Discussion by eggxo** [vegandevs/vegan#614] [Q&A]
Dear author,
When I run RDA anaysis, I find that there are not full environment variables in the  correlation plot, like this: I use 42 environment variables, but there are only 21 in this plot. I have read the A&Q in issue511,but I am still confused about the alias variables:
In my data, the rda only considers the first 21 variables and ignores the rest. Then, I delete the  first 21 variables and perform the rda, it still consider the first 21 variables in the rest variables. But these variables do not have strong correlation, for example: UV and bio1-11.

**Thus, my problems are:
1. If I want to draw the explanatory plot of all 42 variables (like the plots below), what should I do?
2. How could I know variable and its   alias variable? Could I use one of the variables to perform RDA？
3. Also, in rda forward selection, we also meet alias variables, if one is significant, does another is significant too?
4.  When I use the results of RDA to get the intersects of the results of LFMM, there are little intersects. Do u have any suggestions? (my workflow is : use forward selection to select envrionment variables--run rda anaysis and LFMM anaysis--get the intersects)**

Thank you sir very much ! I really appreciate it if u could help me!

### **all variables**
![1](https://github.com/vegandevs/vegan/assets/32643802/e083fea6-d9fd-4da9-9439-8bb7b6e9d43a)
![pade rda1_2 huan](https://github.com/vegandevs/vegan/assets/32643802/4029d56f-d4e5-42e2-932a-c60a53074469)

### **change the sorts**
![2](https://github.com/vegandevs/vegan/assets/32643802/e49df31f-4c1b-41b8-96ac-87023088037c)
![pade rda1_2 huan1](https://github.com/vegandevs/vegan/assets/32643802/c11b570d-3415-4845-871f-5907db5aa3bb)

### **delete bio12-19**
![3](https://github.com/vegandevs/vegan/assets/32643802/f39fee7c-3b9d-4568-878f-d9529a28dfab)
![pade rda1_2 huan3](https://github.com/vegandevs/vegan/assets/32643802/4ae7613d-22d3-42d3-a11e-c7b17aee6c19)

**Comment by gavinsimpson** [comment]
1. Looks like you have only ~24 samples, so no matter what you do you can't estimate the effects of 42 environmental variables. It is simply not possible to estimate more things than you have observations in statistical models like this without some form of regularization and there is not option for that in vegan (nor any other software providing RDA in R, but is for canonical correlation analysis for example, though not in vegan).
2. Aliasing means that 1 or more of your variables is linearly dependent on one or more of the other environmental variables. It doesn't matter what the individual correlations are between pairs of variables are, but whether you can form one or more variables as a linear combination of the other variables. Perhaps once you know `bio1` and `bio6`, `bio5` provides no additional information, hence one of those three will become aliased.
3. I don't understand what you mean by

    > When I use the results of RDA to get the intersects of the results of LFMM, there are little intersects

    What does "intersects" mean? LFMM includes regularization in the form of ridge or lasso penalties, so it doesn't surprise me that it might be able to fit all your variables for reasons I mentioned above. But then you have the issue of interpretting the effects of those variables as the estimates are biased by virtue of the regularization.

4. You'll see the rank defficiency (as that is what it is called; your matrix of environmental variables is not of full rank (42) but of some lower rank and that is what `rda()` is telling you with the aliasing) in `rda()` with forward selection, but only if the one of those variables that is linearly dependent on others in the set are selected for inclusion. We don't recommend doing standard forward selection with inclusion determined by AIC or permutation *p* value. Instead, advice is to fit the full model and proceed only if the omnibus test of that model is statistically significant. Then we use two stopping rules while doing the forward selection:

    1. Does the adjusted $R^2$ of the current model exceed the adjusted $R^2$ of the full model?
    2. Does the variable that improves the model most at this step make a statistically significant improvement to the model?

    The problem you have is that you can't fit this full model because of the linear dependence.
5. You could use `envfit()` to visualise the aliased variables.
